### DISCUSSION TITLE
Stuttering when using Exiled-Exchange-2.

### DISCUSSION THREAD

**Discussion by RogueMuppetArms** [Kvan7/Exiled-Exchange-2#151] [General]
I have found this when I run the Exiled-Exchange-2 app.
I had an AMD 3600 CPU, at first I was thinking that CPU was just not up to the task but that's not the case as I just installed the AMD 7 5700x3d and I still have the same problem.
Just wanted to let you know this as it could be a problem for others.

**Comment by Sv1ngen** [comment]
I have the exact same issue. Im on a AMD 5600x CPU and a 7900 GRE GRP and as soon as i open Exiled Exchange 2 the game stops using my 100 FPS lock and goes beoynd and starts to stutter.

**Reply by Sv1ngen** [reply]
Tried Sidekick just for fun, when i press ctrl+d Poe2 windows looses focus and FPS gets wonky exactly as exile exhange does, the difference is that when i close the pricecheck windows poe2 windows gets focus again and fps is stable. 
Is there anything you can do so that Exile Exchange dosent steal focus from game window when you are not pricechecking anything?

**Reply by Kvan7** [reply]
the option would be to enable `--no-overlay` [cli flag](https://kvan7.github.io/Exiled-Exchange-2/cmd-flags.html), but that is probably not what you are looking for

**Comment by Kvan7** [comment]
This _may_ be because the game loses focus so it gets less performance? I'm really not sure on the hardware problem stuff sorry.

**Reply by maxdd** [reply]
I do have the same issue, could it be the "fps while in background" being too low? I haven't tried to change it back but currently i have a value lower than 10fps

EDIT:
That didn't change a thing

**Reply by maxdd** [reply]
@Kvan7 could you build EE2 with [this](https://github.com/Kvan7/Exiled-Exchange-2/blob/d02f086f19643cb7f4bf7694b9b98e1c0b0b0b69/main/src/main.ts#L24) enabled for win32 instead?
The [demo](https://github.com/SnosMe/electron-overlay-window/blob/master/src/demo/electron-demo.ts#L5) is reporting an issue for linux but i'm wondering if it is really needed on windows

**Reply by Kvan7** [reply]
That line _does_ appear to run when building for windows machines?

**Reply by maxdd** [reply]
I'm no npm/nodejs/ts expert but my understanding is that that line of code is executed everytime we launch EE2 right? (it has nothing to do with building in windows). It is in the main.ts and if the host is different from macos (e.g. linux and win32) it will disable HW accel right?

EDIT:
I was able to build a version which i will test this evening

**Reply by Kvan7** [reply]
sorry, then it runs every time for windows since `darwin` is mac. I just don't know if you think this should or should not run

**Reply by maxdd** [reply]
My reasoning is that if you disable hw acceleration everything is allocated to the cpu which might explain the lagging and the increased overload. I have read about ppl switching to faster CPU (Ryzen 5000 series) without any luck though so I'm just exploring. There was another guy mentioning that it has something to do with his monitor Hz as if gsync (or freesync) is not working properly.
So, if for Mac it is ok to have it, I don't see why not for windows since for Linux it was removed because of a bug

**Reply by maxdd** [reply]
i did the test, it didnt resolve it. The only solution for me is to use the --no-overlay and use the browser by tabbing

**Reply by maxdd** [reply]
I have confirmed it is related to freesync. If i turn it off from the monitor itself i have full fps + EE2 running.
I guess in the APT repo there are similar reports and it might be worth discussing there as well

**Comment by vincent-v-c** [comment]
I notice my GSYNC stops working as soon as I launch Exiled Exchange, i did not have this issue in the early versions.

**Reply by maxdd** [reply]
was it stuttering in early versions?
https://github.com/SnosMe/electron-overlay-window/issues/25

**Comment by donlooney** [comment]
I have the same problem on my system. AMD Ryzen 7 5700X & AMD RX 6800 XT Freesync 144 Hz | Windows 11 latest.
Using windowed fullscreen mode. 2 monitor setup. EE2 is causing massive micro stutter.

**Reply by maxdd** [reply]
That's freesync, not sure whether the driver or the monitor but if you turn it off the stutter also goes away

**Reply by donlooney** [reply]
Thanks for the info.
